sdk/containerservice/Azure.ResourceManager.ContainerService/src/Custom/LongRunningOperation/ContainerServiceArmOperation.cs

```diff
@@ -65,5 +99,58 @@ internal ContainerServiceArmOperation(ClientDiagnostics clientDiagnostics, HttpP
 
         /// <inheritdoc />
         public override ValueTask<Response> WaitForCompletionResponseAsync(TimeSpan pollingInterval, CancellationToken cancellationToken = default) => _operation.WaitForCompletionResponseAsync(pollingInterval, cancellationToken);
+
+        // TODO: use this class as a temporary workaround before https://github.com/Azure/autorest.csharp/issues/2231 is resolved so that we can deserialize to the original HttpWebResponseImplementation in LRO helper classes.
+        // No all methods and properties are implemented/populated as it's just for prototype.
+        internal class ContainerServiceResponse: Response
```

Comment:
This class is temporary for prototype purpose as we cannot deserialize the response to the abstract class of `Azure.Core.Response`. The final solution is likely to depend on https://github.com/Azure/autorest.csharp/issues/2231. Then we can deserialize to the original `HttpWebResponseImplementation`.